infer/provider.go

```diff
@@ -140,7 +142,21 @@ func Wrap(provider p.Provider, opts Options) p.Provider {
 
 	config := opts.Config
 	if config != nil {
-		provider.Configure = config.configure
+		if prev := provider.Configure; prev != nil {
+			provider.Configure = func(ctx p.Context, req p.ConfigureRequest) error {
+				err := config.configure(ctx, req)
```

Comment:
Since we're calling both `Configure`'s in this case, it's not so much delegation as wrapping, isn't it?

Wouldn't it be more common to call the outermost layer first, then the more specific one last?

Reply:
> Since we're calling both `Configure`'s in this case, it's not so much delegation as wrapping, isn't it?

Yeah, it's a wrapping call. Delegate was a bad choice of PR name. The function this is in is actually called `Wrap`.

> Wouldn't it be more common to call the outermost layer first, then the more specific one last?

That is what is happening. `config.configure` is the inferred "outer" layer and `prev` is the "inner" layer.